### ISSUE TITLE
Ventura bottling

### ISSUE THREAD

**Issue by Bo98** [Homebrew/homebrew-core#113717]
This is a tracking issue on Ventura bottling process and will be used as a dump for build failures.

If you are a user of Homebrew, this isn't where to seek help or report issues. You can get support via the [discussions forum](https://github.com/orgs/Homebrew/discussions) or if you believe you are facing a reproducible bug then you can [open a new issue](https://github.com/Homebrew/homebrew-core/issues/new/choose).

If you would like to contribute to Homebrew, [open pull requests to fix build failures](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md#to-contribute-a-fix-to-the-foo-formula). If the build works on Ventura but there's no bottle, that likely means we just haven't got around to bottling that particular formula yet and we will get there eventually!

---

**Maintainers: please read all of this before continuing!**

As is now standard, we will be bottling for both x86_64 and arm64. However, **Intel CI is not yet available so please bottle for 13-arm64 only at this time**. There is no definitive ETA for Intel CI as it is outside our control, apologies.

To start a bottle build, run `brew dispatch-build-bottle --macos=13-arm64 --issue=??? --upload <formula>` or use the equivalent UI [over at the Actions workflow runs page](https://github.com/Homebrew/homebrew-core/actions/workflows/dispatch-build-bottle.yml).

**Please do not bottle something until all its dependencies have been bottled and their associated bottle commits have been pushed to master.**

Please take care to watch the queue over on [our dashboard](https://orchestrator.brew.sh) (available to maintainers only) or on the [Actions workflow runs page](https://github.com/Homebrew/homebrew-core/actions/workflows/dispatch-build-bottle.yml). Avoid running duplicated runs and and avoid excessively adding to a queue that may already be massive.

If a build failure occurs, logs will be posted as comments here (if you configured it to do so when starting the build). Please check those comments before dispatching a build for a formula that may have already been attempted. To help with this I have created a tracker page to iterate over the comments and extract failiing build logs for formulae that have not had a successful bottle build yet: **URL TBC** (available to maintainers only; page may be slow so avoid refreshing too much). Items will disappear from that webpage when a bottle has been successfully built for _both_ x86_64 and arm64.

Shortly, I will start by focussing on popular formula with a high number of dependents while closely monitoring to see if everything is working ok. Feel free to join in after a few hours when it's clear it's all working.

**Comment by Bo98** [comment]
yasm test likely only ever worked under Rosetta 2, which our new 13-arm64 runners don't have. yasm doesn't support producing arm64 binaries.

**Comment by cho-m** [comment]
yasm/nasm dependencies usually belong in `on_intel` sections (e.g. #113741), which would lower priority of bottling `yasm`.

`openldap` looks like result of removing `groff` (#105682). Can't test it, but #113742 might fix build.

**Comment by carlocab** [comment]
> :x: @cho-m bottle request for libxcb [failed](https://github.com/Homebrew/homebrew-core/actions/runs/3302060227).

Did `libxcb` find its way to the `curl` dep tree?

https://github.com/Homebrew/homebrew-core/actions/runs/3302060227/jobs/5448377944#step:7:47

**Comment by cho-m** [comment]
> Did `libxcb` find its way to the `curl` dep tree?
> 
> https://github.com/Homebrew/homebrew-core/actions/runs/3302060227/jobs/5448377944#step:7:47

Sadly, I think `openldap` was part of `curl` dep tree. We may need to disable manpages.

**Comment by carlocab** [comment]
Can I get a review on #111079? I'd like to get this in before we start bottling z3 dependents so that we get a properly versioned `libz3`.

**Comment by Bo98** [comment]
> xgcc: warning: could not understand version '13.00.00'

Is there a reason we need to patch GCC for a new macOS every year?

**Comment by Bo98** [comment]
Upstream patch for assimp: https://github.com/assimp/assimp/commit/5a89d6fee138f8bc979b508719163a74ddc9a384

**Comment by Bo98** [comment]
https://gitlab.haskell.org/ghc/ghc/-/commit/c2043b0a2132d1f8b6a22f3daacaaf6fb194db17.

Fixed in 9.2.5 (not released) and 9.4.2.

It probably also affect the bootstraps, which would be a concern we'll need to think how to resolve.

**Comment by chenrui333** [comment]
> ❌ @chenrui333 bottle request for amfora [failed](https://github.com/Homebrew/homebrew-core/actions/runs/3303595137).

Looks like the terminal does not run properly on Ventura

**Comment by Bo98** [comment]
> ❌ @fxcoudert bottle request for libconfig [failed](https://github.com/Homebrew/homebrew-core/actions/runs/3304520966).

For errors like this one, instead of depending on `texinfo` let's try use the proper tarball which should mean autotools are not needed: https://github.com/hyperrealm/libconfig/releases/download/v1.7.3/libconfig-1.7.3.tar.gz

**Comment by carlocab** [comment]
> :x: @cho-m bottle request for r [failed](https://github.com/Homebrew/homebrew-core/actions/runs/3304722052).

```
  devQuartz.c:660:23: error: 'ATSFontFindFromName' is unavailable
1676
              atsFont = ATSFontFindFromName(cfFontName, kATSOptionFlagsDefault);
1677
                        ^
```

**Comment by fxcoudert** [comment]
Fixing the yasm test to run well without Rosetta 2: https://github.com/Homebrew/homebrew-core/pull/113824

**Comment by carlocab** [comment]
> ❌ @fxcoudert bottle request for libvpx [failed](https://github.com/Homebrew/homebrew-core/actions/runs/3308290303).
> 
> ```
> ld: unknown option: --no-undefined
> ```

`configure` is misdetecting the system as `generic-gnu`. Can be worked around as in #88518.

**Comment by Bo98** [comment]
> > ❌ @fxcoudert bottle request for libvpx [failed](https://github.com/Homebrew/homebrew-core/actions/runs/3308290303).
> > ```
> > ld: unknown option: --no-undefined
> > ```
> 
> `configure` is misdetecting the system as `generic-gnu`. Can be worked around as in #88518.

One of those projects that added support for darwin20, darwin21, but not darwin22...

https://github.com/webmproject/libvpx/blob/5245f6e9cb7e6bb68ab45fe4d8b00bc9b16857e1/build/make/configure.sh#L794

**Comment by carlocab** [comment]
> One of those projects that added support for darwin20, darwin21, but not darwin22...
> 
> https://github.com/webmproject/libvpx/blob/5245f6e9cb7e6bb68ab45fe4d8b00bc9b16857e1/build/make/configure.sh#L794

Wait till you see what they pass to `sed`. https://github.com/webmproject/libvpx/blob/5245f6e9cb7e6bb68ab45fe4d8b00bc9b16857e1/build/make/configure.sh#L796

**Comment by Bo98** [comment]
Darwin 30 in 2030 is going to be fun.

**Comment by Bo98** [comment]
> ❌ @fxcoudert bottle request for bsdmake [failed](https://github.com/Homebrew/homebrew-core/actions/runs/3308635030).

Oh, we still have XQuartz references?

**Comment by Bo98** [comment]
> ❌ @cho-m bottle request for hashcat [failed](https://github.com/Homebrew/homebrew-core/actions/runs/3310438778).
> 
> ```
> clGetDeviceIDs(): CL_INVALID_VALUE
> No devices found/left.
> ```

Oh, I should have mentioned this: our 13-arm64 machines no longer support OpenCL. This is not a macOS 13 issue but is a result of moving to virtualised machines. OpenCL is not supported on Apple Silicon VMs.

**Comment by Bo98** [comment]
> ❌ @fxcoudert bottle request for tor [failed](https://github.com/Homebrew/homebrew-core/actions/runs/3314758050).
> 
> `ld: in libtor.a(__.SYMDEF SORTED), archive member '__.SYMDEF SORTED' with length 496 is not mach-o or llvm bitcode file 'libtor.a' for architecture arm64`

What MacPorts did: https://github.com/macports/macports-ports/commit/4b4383b53db935a6dc2a6deb253cce1dfa735931

**Comment by cho-m** [comment]
> ❌ @chenrui333 bottle request for bandwhich [failed](https://github.com/Homebrew/homebrew-core/actions/runs/3304091197).

> ❌ @fxcoudert bottle request for flux [failed](https://github.com/Homebrew/homebrew-core/actions/runs/3316250498).

> ❌ @cho-m bottle request for influxdb [failed](https://github.com/Homebrew/homebrew-core/actions/runs/3326058446).

> ❌ @cho-m bottle request for spotify-tui [failed](https://github.com/Homebrew/homebrew-core/actions/runs/3326966639).

I think most of these failures were introduced by updating to `rust` 1.64.0 (#111434). Probably should have followed up on failures upstream and tracked corresponding fixes.

`flux` is at least fixed in new release (https://github.com/influxdata/flux/releases/tag/v0.187.0). `influxdb` will need a new release with its version of flux updated to build.

**Comment by chenrui333** [comment]
> ❌ @chenrui333 bottle request for clair [failed](https://github.com/Homebrew/homebrew-core/actions/runs/3396527916).

checksum mismatch, #114738

**Comment by chenrui333** [comment]
> ❌ @chenrui333 bottle request for wasm-pack [failed](https://github.com/Homebrew/homebrew-core/actions/runs/3304412283).
> 
> ```
> ==> /opt/homebrew/Cellar/wasm-pack/0.10.3/bin/wasm-pack new hello-wasm
>   [INFO]: Installing cargo-generate...
>    Generating a new rustwasm project with name 'hello-wasm'...
>   Error: Running cargo-generate
>   Caused by: Bad CPU type in executable (os error 86)
> ```

relates to https://github.com/rustwasm/wasm-pack/issues/1098

**Comment by chenrui333** [comment]
> ❌ @fxcoudert bottle request for cabocha [failed](https://github.com/Homebrew/homebrew-core/actions/runs/3405348947).

The source artifact is not available, raise this up to help@mirrorservice.org

**Comment by fxcoudert** [comment]
@Moisan most of the bottles you launched are known failures (see above)

**Comment by chenrui333** [comment]
> ❌ @fxcoudert bottle request for gopass [failed](https://github.com/Homebrew/homebrew-core/actions/runs/3494979991).
> 
> test failure:
> 
> ```
> runtime: nameOff 0x34312e31 base 0x106667b0 not in ranges:
>   	types 0x10679000 etypes 0x107ffce8
>   fatal error: runtime: name offset base pointer out of range
> ```

Should be good now, but interestingly, it did not turn green on the UI

**Comment by chenrui333** [comment]
> ❌ @chenrui333 bottle request for pandoc [failed](https://github.com/Homebrew/homebrew-core/actions/runs/3561508444).

Filed an upstream issue, https://github.com/jgm/pandoc/issues/8457

**Comment by fxcoudert** [comment]
Closing the issue. At this stage most bottles are built, remaining top issues are tracked in https://github.com/Homebrew/homebrew-core/issues/116910
